tests/model_registry/rbac/test_mr_rbac_sa.py

```diff
@@ -0,0 +1,145 @@
+            assert http_error.body is not None, "HTTPError should have a response object"
+            LOGGER.info(f"Received expected HTTP error: Status Code {http_error.status}")
+            assert http_error.status == 403, f"Expected HTTP 403 Forbidden, but got {http_error.status}"
+            LOGGER.info("Successfully received expected HTTP 403 status code.")
+
+        except Exception as e:
+            LOGGER.error(f"Received unexpected error during 'access denied' check: {e}", exc_info=True)
+            pytest.fail(f"'Access denied' check failed unexpectedly: {e}")
+
+    @pytest.mark.smoke
+    # Use fixtures for SA/NS/Token AND the RBAC Role/Binding
+    @pytest.mark.usefixtures("sa_namespace", "service_account", "mr_access_role", "mr_access_role_binding")
+    def test_service_account_access_granted(
+        self: Self,
+        sa_token: str,  # Get token for the test SA
+        model_registry_instance_rest_endpoint: str,  # REST endpoint fixture
+        # mr_access_role and mr_access_role_binding are activated by @usefixtures
+    ):
+        """
+        Verifies SA access is GRANTED via REST after applying Role and RoleBinding fixtures.
+        """
+        LOGGER.info("--- Starting RBAC Test: Access Granted ---")
+        LOGGER.info(f"Targeting Model Registry REST endpoint: {model_registry_instance_rest_endpoint}")
+        LOGGER.info("Applied RBAC Role/Binding via fixtures. Expecting access GRANT.")
+
+        try:
+            client_args = build_mr_client_args(
+                rest_endpoint=model_registry_instance_rest_endpoint, token=sa_token, author="rbac-test-granted"
+            )
+            LOGGER.debug(f"Attempting client connection with args: {client_args}")
+
+            # Instantiate the client - this should now succeed or fail for non-auth reasons
+            mr_client_success = ModelRegistryClient(**client_args)
+            assert mr_client_success is not None, "Client initialization failed after granting permissions"
+            LOGGER.info("Client instantiated successfully after granting permissions.")
+
+        except ForbiddenException as e:
+            # If we get an HTTP error here, it's unexpected, especially 403
+            LOGGER.error(
+                f"Received unexpected HTTP error after granting access: {e.status if e.body else 'No Response'} - {e}",
+                exc_info=True,
+            )
+            if e.body is not None and e.status == 403:
+                pytest.fail(f"Still received HTTP 403 Forbidden after applying Role/RoleBinding: {e}")
+            else:
+                pytest.fail(f"Client interaction failed with HTTP error after granting permissions: {e}")
+
+        except Exception as e:
+            LOGGER.error(f"Received unexpected general error after granting access: {e}", exc_info=True)
+            pytest.fail(f"Client interaction failed unexpectedly after granting permissions: {e}")
+
+        LOGGER.info("--- RBAC Test Completed Successfully ---")
```

Comment:
Do you need this part? Any exception here is bad and should be raised as they arise.